lib/salus/report.rb

```diff
@@ -222,9 +224,22 @@ def publish_report(directive)
       end
     end
 
+    def satisfies_filter?(directive)
+      return true if @report_filter == 'all'
+      return false if @report_filter == 'none'
+
+      recovered_values = @report_filter.split(':')
+      filter_key = recovered_values[0]
+      filter_value = recovered_values[1]
+
+      directive.key?(filter_key) && (
+        directive[filter_key] == filter_value || directive[filter_key] == '*'
```

Comment:
Shouldn't you check if something starts with `filter_value`, instead of `==`?
if your unit test didn't catch this, then you may want to update your unit tests.

Reply:
If I remember correctly, there was a related bug when you were doing the regex stuff.  The regex match was not looking for lines that starts with a pattern, but was looking for lines that included the pattern.

Reply:
I actually started with using `start_with?()` but decided to use `==` to prevent the filter from including things we don't want. For example, `sarif` is a prefix of `sarif_diff`  and `http://example.com/dir` is a prefix of `http://example.com/dir/subdir` so I think the added strictness helps.

Reply:
hmm, what if the pattern is `url:http`, then it will not match anything (unless the url really is just `http`, right?